### PR TITLE
Add example of attaching transitions and transitioning on build setting

### PR DESCRIPTION
**Pull request by juliexxia**
Work towards #154 

This one actually covers quite a bit of stuff, but it made sense to me to keep it all together - let me know if you think it's too dense

### CODE REVIEW

File: rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    return {"//starlark_configurations/attaching-transitions-to-rules:color": "red"}
+
+red_transition = transition(
+    implementation = _red_impl,
+    inputs = [],
+    outputs = ["//starlark_configurations/attaching-transitions-to-rules:color"],
+)
+
+def _impl(ctx):
+    # Access the value of "//experimental/users/juliexxia:color" for the target (blue).
```
Comment (gregestren) [inline rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl:29]: Update the package name?

File: rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    print("shirt color: " + ctx.attr._color[BuildSettingInfo].value)
+    # Access the value of "//experimental/users/juliexxia:color" for the transitioned dep (red).
```
Comment (gregestren) [inline rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl:31]: Update the package name?

File: rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    # Note that you have to index by [0] here for the transitioned dep and you don't need to
+    # do so below - this is because attribute-attached transitions can transition to multiple
+    # new configurations so you must specify which one you want.
+    print("sleeve color: " + ctx.attr.sleeve[0][BuildSettingInfo].value)
+    # Access the value of "//experimental/users/juliexxia:color" for the non-transitioned dep (blue).
```
Comment (gregestren) [inline rules/starlark_configurations/attaching-transitions-to-rules/defs.bzl:36]: And here?

**Comment by juliexxia** [comment]
Thanks for those catches (: all done ptal!

**Pull request merged**